cpp/src/io/parquet/page_data.cu

```diff
@@ -328,23 +330,25 @@ CUDF_KERNEL void __launch_bounds__(decode_block_size)
     int target_pos;
     int src_pos = s->src_pos;
 
-    if (t < out_thread0) {
-      target_pos = min(src_pos + 2 * (decode_block_size - out_thread0),
-                       s->nz_count + (decode_block_size - out_thread0));
+    if (warp.meta_group_rank() < out_warp_id) {
+      target_pos =
+        cuda::std::min<int32_t>(src_pos + 2 * (decode_block_size - (out_warp_id * warp.size())),
+                                s->nz_count + (decode_block_size - (out_warp_id * warp.size())));
```

Comment:
another very optional one: IMO would be great if `decode_block_size - (out_warp_id * warp.size())` was a local variable.

Reply:
Done!